File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
    
        res = [0,1,2]
        
        i = 3
        
        while i <= n:
            res.append(res[i-1] + res[i-2])
            i+=1
            
        return res[-1]
            
        

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 2) {
            return n;
        }

        int arr[2];
        int i = 2;
        
        arr[0] = 1;
        arr[1] = 2;

        while (i < n) {
            int tmp = arr[1];
            arr[1] = arr[0] + arr[1];
            arr[0] = tmp;
            i++;
        }

        return arr[1];
    }
};